### PR TITLE
UIBULKED-298: Group Item record properties using optgroup component

### PR DESCRIPTION
**Pull request by UladzislauKutarkin**
## Purpose

[UIBULKED-298](https://issues.folio.org/browse/UIBULKED-298)
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by UladzislauKutarkin** [comment]
https://github.com/folio-org/ui-bulk-edit/assets/72550466/5d77a31b-1a3d-41f9-bdc2-17281f1f247f

**Review by zburke — CHANGES_REQUESTED** [review]
The `renderOptions` function itself looks good but the `category` attribute of the updated `getItemsOptions` elements must be i18n'ed. 

Please obey lint instead of turning it off :)

File: src/constants/selectOptions.js [hunk 1]
```diff
@@ -185,11 +185,13 @@ export const getItemsOptions = (formatMessage) => [
     value: OPTIONS.TEMPORARY_LOCATION,
     label: formatMessage({ id: 'ui-bulk-edit.layer.options.temporaryLocation' }),
     disabled: false,
+    category: 'Location',
```
[the remark this answers is not in the document]
Comment (UladzislauKutarkin) [reply src/constants/selectOptions.js:188]: fixed

File: src/constants/selectOptions.js [hunk 2]
```diff
@@ -185,11 +185,13 @@ export const getItemsOptions = (formatMessage) => [
...
   },
   {
     value: OPTIONS.PERMANENT_LOCATION,
     label: formatMessage({ id: 'ui-bulk-edit.layer.options.permanentLocation' }),
     disabled: false,
+    category: 'Location',
```
[the remark this answers is not in the document]
Comment (UladzislauKutarkin) [reply src/constants/selectOptions.js:195]: fixed

File: src/constants/selectOptions.js [hunk 3]
```diff
@@ -200,11 +202,13 @@ export const getItemsOptions = (formatMessage) => [
     value: OPTIONS.TEMPORARY_LOAN_TYPE,
     label: formatMessage({ id: 'ui-bulk-edit.layer.options.temporaryLoanTypeLabel' }),
     disabled: false,
+    category: 'Loan type',
```
[the remark this answers is not in the document]
Comment (UladzislauKutarkin) [reply src/constants/selectOptions.js:207]: fixed

File: src/utils/filters.js [hunk 4]
```diff
@@ -34,3 +34,28 @@ export const getDefaultCapabilities = (view, perms) => {
...
+      if (!categoryMap[item.category]) {
+        categoryMap[item.category] = [];
+      }
+      categoryMap[item.category].push(item);
+    } else {
+      convertedArray.push(item);
+    }
+  }
+
+  // eslint-disable-next-line guard-for-in
```
[the remark this answers is not in the document]
Comment (UladzislauKutarkin) [reply src/utils/filters.js]: I agree, I fixed as eslint expecting it.

**Review by zburke — APPROVED** [review]

**Review by vashjs — APPROVED** [review]

**Pull request merged**